docs/09_tutorials/01_binary-extension.md

```diff
@@ -0,0 +1,720 @@
+-           "type": "name"
+        }
+    ]
+}
+```
+
+Note the `$` after the types now; this indicates that this type is an `eosio::binary_extension` type field.
+```diff
+{
+    "name": "secondary_key",
++   "type": "name$"
+-   "type": "name"
+}
+```
+
+```diff
+{
+    "name": "_binary_extension_key",
++   "type": "name$"
+-   "type": "name"
+}
+```
+
+Now, let's upgrade the contract again and try to read/write from/to our table:
+
+```
+~/binary_extension_contract $ cleos set contract eosio ./
+```
+
+```
+Reading WASM from /Users/john.debord/binary_extension_contract/binary_extension_contract.wasm...
+Publishing contract...
+executed transaction: 497584d4e43ec114dbef83c134570492893f49eacb555d0cd47d08ea4a3a72f7  4696 bytes  648 us
+#         eosio <= eosio::setcode               "0000000000ea30550000cb6a0061736d01000000018f011760000060017f0060027f7f0060037f7f7f017f6000017e60017...
+#         eosio <= eosio::setabi                "0000000000ea305581030e656f73696f3a3a6162692f312e310006076d6f646966797000020b7072696d6172795f6b65790...
+warning: transaction executed locally, but may not be confirmed by the network yet
+```
+
+```
+~/binary_extension_contract $ cleos push action eosio printbyp '{"primary_key":"eosio.name"}' -p eosio
+```
+
+```
+executed transaction: 6108f3206e1824fe3a1fdcbc2fe733f38dc07ae3d411a1ccf777ecef56ddec97  104 bytes  224 us
+#         eosio <= eosio::printbyp              {"primary_key":"eosio.name"}
+[(eosio,printbyp)->eosio]: CONSOLE OUTPUT BEGIN =====================
+`printbyp` executing.
+`_primary_key`: eosio.name found; printing.
+{eosio.name, nothin}
+`printbyp` finished executing.
+[(eosio,printbyp)->eosio]: CONSOLE OUTPUT END   =====================
+warning: transaction executed locally, but may not be confirmed by the network yet
+```
+
+```
+~/binary_extension_contract $ cleos push action eosio regpkey '{"primary_key":"eosio.name2"}' -p eosio
+```
+
+```
+executed transaction: 75a135d1279a9c967078b0ebe337dc0cd58e1ccd07e370a899d9769391509afc  104 bytes  227 us
+#         eosio <= eosio::regpkey               {"primary_key":"eosio.name2"}
+[(eosio,regpkey)->eosio]: CONSOLE OUTPUT BEGIN =====================
+`regpkey` executing.
+`_primary_key`: eosio.name2 not found; registering.
+`regpkey` finished executing.
+[(eosio,regpkey)->eosio]: CONSOLE OUTPUT END   =====================
+warning: transaction executed locally, but may not be confirmed by the network yet
+```
+
+Nice! The smart contract is now backwards compatible for the future use of its tables and/or actions.
+
+<hr>
+
+Just keep these simple rules in mind when upgrading a smart contract.
+If you are adding a new field to a struct currently in use by a `eosio::multi_index` be **SURE** to:
+- add the field at the end of the struct,
+- and wrap the type using an `eosio::binary_extension` type.
+
+Also, there are a few restrictions you have to be aware of which are outlined [here](https://github.com/EOSIO/eos/issues/5600).
```

Comment:
We should not link to the issue, we should instead transpose this information to documentation. Issues are not a place for documentation.